### DISCUSSION TITLE
SSSD full Support

### DISCUSSION THREAD

**Discussion by HilkopterBob** [goauthentik/authentik#10105] [Ideas]
Good Morning!

I currently evaluate authentik for my company. I saw your docs about SSSD support and that you don't fully utilize the standard, eg. no auto sudo, no auto mounts etc. 

That forces users like me to create rather crooked setups where I manage User-IDs in Free IPA, from which Authentik grabs them via LDAP and manages ID for Web apps and possibly Remote Services.

Nothing wrong with this setup but I see untapped market potential if Authentik centralizes not only WebID'ing but also SSSD Management. Essentially upgrading Authentik to an AD-Controller respectively.

**Comment by Fuseteam** [comment]
i managed to set up "auto sudo" as follows:

in the UI create a group "sudo"
and make the gid "27"
add users to this group et voila auto sudo

for reference:
![image](https://github.com/user-attachments/assets/b4e65391-00fb-4b4e-88c1-6aca39ddb1d6)

**Reply by HilkopterBob** [reply]
this looks promising, thank you! i will try this approach and report back later. Do you mind explaining your setup and environment more in depth?

**Reply by Fuseteam** [reply]
it is a fairly standard setup, what would like to know about the environment?

**Reply by HilkopterBob** [reply]
your client side sssd setup and how it is interfacing with authentik mostly, thank you for your work by the way

**Reply by Fuseteam** [reply]
for that i used this /etc/sssd/sssd.conf
```
[nss]
reconnection_retries = 3

[sssd]
config_file_version = 2
reconnection_retries = 3
domains = authentik.example.com
# uncomment for centos
# services = nss, pam, ssh

[pam]
reconnection_retries = 3

[domain/authentik.example.com]
cache_credentials = True
id_provider = ldap
chpass_provider = ldap
auth_provider = ldap
access_provider = ldap
ldap_uri = ldaps://authentik.example.com

ldap_schema = rfc2307bis
ldap_search_base = dc=authentik,dc=example,dc=com
ldap_user_search_base = ou=users,dc=authentik,dc=example,dc=com
ldap_group_search_base = dc=authentik,dc=example,dc=com

ldap_user_object_class = user
ldap_user_name = cn
ldap_group_object_class = group
ldap_group_name = cn

# Optionally, filter logins to only a specific group
ldap_access_order = filter
ldap_access_filter = memberOf=cn=authentik Admins,ou=groups,dc=authentik,dc=example,dc=com

ldap_default_bind_dn = cn=serviceaccount,ou=users,dc=authentik,dc=example,dc=com
ldap_default_authtok = supersecretpassword
```
